lib/trento_web/controllers/v1/activity_log_controller.ex

```diff
@@ -11,15 +11,110 @@ defmodule TrentoWeb.V1.ActivityLogController do
   operation :get_activity_log,
     summary: "Fetches the Activity Log entries.",
     tags: ["Platform"],
+    parameters: [
+      first: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :integer},
+        required: false
+      ],
+      last: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :integer},
+        required: false
+      ],
+      after: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      before: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      from_date: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      to_date: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      actor: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :array},
+        required: false
+      ],
+      type: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :array},
+        required: false
+      ]
+    ],
     responses: [
       ok: {"Activity Log settings fetched successfully", "application/json", Schema.ActivityLog}
     ]
 
-  def get_activity_log(conn, _) do
-    with activity_log_entries <- ActivityLog.list_activity_log() do
+  def get_activity_log(conn, params) do
+    parsed_params =
+      case params do
+        [] -> %{}
+        params -> parse_params(params)
+      end
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** I see some downsides in doing the parsing step in the controller:
* we leak some implementation details (Flop's filters shape) to the integration point (the controller);
* we handle 2 different interfaces for the same problem (the API of the controller and the signature of the context);
* we are _invited_ to use Flop's internals in our test (which we should avoid, our tests should use as plain data as possible).

I think we can have a better result by moving the parsing step into the context: 
* our integration with Flop will be segregated in the core domain without leaking into the integration layer;
* we can test filters and pagination directly against the context (no need to use an intermediate interface in tests).

What do you think?


‏